commit afd962a9e8708c571c5c0c4a6d098f931742c229
Author: Vitaly Mayatskikh <v.mayatskih@gmail.com>
Date:   Wed Jul 30 13:30:14 2008 +0200

    x86: wrong register was used in align macro
    
    New ALIGN_DESTINATION macro has sad typo: r8d register was used instead
    of ecx in fixup section. This can be considered as a regression.
    
    Register ecx was also wrongly loaded with value in r8d in
    copy_user_nocache routine.
    
    Signed-off-by: Vitaly Mayatskikh <v.mayatskih@gmail.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/arch/x86_64/lib/copy_user.S linux-source-2.6.18/arch/x86_64/lib/copy_user.S
--- linux-source-2.6.18.orig/arch/x86_64/lib/copy_user.S	2008-07-27 16:46:41.000000000 -0600
+++ linux-source-2.6.18/arch/x86_64/lib/copy_user.S	2008-08-12 00:13:09.000000000 -0600
@@ -52,7 +52,7 @@
 	jnz 100b
 102:
 	.section .fixup,"ax"
-103:	addl %r8d,%edx			/* ecx is zerorest also */
+103:	addl %ecx,%edx			/* ecx is zerorest also */
 	jmp copy_user_handle_tail
 	.previous
 
